Applications/TransferLearning/Draw_Classification/jni/main.cpp

```diff
@@ -39,184 +39,201 @@
+#define NUM_DATA_PER_LABEL 5
 
-/**
- * @brief     Number of Test Set
- */
+/** Size of each label (number of label categories) */
+#define LABEL_SIZE 3
+
+/** Size of each input */
+#define INPUT_SIZE 128
+
+/** Number of test data points */
 #define TOTAL_TEST_SIZE 8
 
-/**
- * @brief     Max Epochs
- */
-#define ITERATION 1000
+/** Total number of data points in an epoch */
+#define EPOCH_SIZE LABEL_SIZE *NUM_DATA_PER_LABEL
 
-using namespace std;
+/** Max Epochs */
+#define EPOCHS 1000
 
-/**
- * @brief     location of resources ( ../../res/ )
- */
-string data_path;
+/** labels values */
+const std::string label_names[LABEL_SIZE] = {"happy", "sad", "soso"};
+
+/** Vectors containing the training data */
+std::vector<std::vector<float>> inputVector, labelVector;
+unsigned int iteration = 0;
 
 /**
  * @brief     step function
  * @param[in] x value to be distinguished
  * @retval 0.0 or 1.0
  */
-float stepFunction(float x) {
-  if (x > 0.9) {
-    return 1.0;
-  }
-
-  if (x < 0.1) {
-    return 0.0;
-  }
+// float stepFunction(float x) {
+//   if (x > 0.9) {
+//     return 1.0;
+//   }
+//
+//   if (x < 0.1) {
+//     return 0.0;
+//   }
+//
+//   return x;
+// }
+
+struct TFLiteData {
+  tflite::ops::builtin::BuiltinOpResolver resolver;
+  std::unique_ptr<tflite::Interpreter> interpreter;
+  std::unique_ptr<tflite::FlatBufferModel> model;
+  std::string data_path;
 
-  return x;
-}
+  int output_number_of_pixels;
+  int inputDimReq[MAX_DIM];
+};
 
-/**
- * @brief     Get Feature vector from tensorflow lite
- *            This creates interpreter & inference with ssd tflite
- * @param[in] filename input file path
- * @param[out] feature_input save output of tflite
- */
-void getFeature(const string filename, vector<float> &feature_input) {
+void setupTensorflowLiteModel(const std::string &data_path,
+                              TFLiteData &tflite_data) {
   int input_size;
   int output_size;
-  int *output_idx_list;
-  int *input_idx_list;
-  int inputDim[4];
-  int outputDim[4];
-  int input_idx_list_len = 0;
-  int output_idx_list_len = 0;
+  int len;
+  int outputDim[MAX_DIM];
+
+  tflite_data.data_path = data_path;
   std::string model_path = data_path + "ssd_mobilenet_v2_coco_feature.tflite";
-  std::unique_ptr<tflite::FlatBufferModel> model =
+  tflite_data.model =
     tflite::FlatBufferModel::BuildFromFile(model_path.c_str());
+  if (tflite_data.model == NULL)
+    throw std::runtime_error("Unable to build model from file");
 
-  assert(model != NULL);
-  tflite::ops::builtin::BuiltinOpResolver resolver;
-  std::unique_ptr<tflite::Interpreter> interpreter;
-  tflite::InterpreterBuilder(*model.get(), resolver)(&interpreter);
+  tflite::InterpreterBuilder(*tflite_data.model.get(),
+                             tflite_data.resolver)(&tflite_data.interpreter);
 
-  input_size = interpreter->inputs().size();
-  output_size = interpreter->outputs().size();
+  if (tflite_data.interpreter->AllocateTensors() != kTfLiteOk)
```

Comment:
Need to add corresponding error handling in the caller. 
The static checker checks if an error is thrown to the very top level :(

Reply:
Added try catch to the top level for now.
In the upcoming PR, this code will be replaced with nnstreamer single API.